### PR TITLE
Enable coverage and docs

### PR DESCRIPTION
**Pull request by DieBauer**
Addresses two points of https://github.com/typelevel/cats/issues/2347

Enable scoverage for 2.13 and enable scaladoc generation for 2.13. 

One caveat with scoverage is that, for scala 2.13.1 (which Cats currently doesn't use), there are issues as described here https://github.com/scoverage/sbt-scoverage/issues/295 and blocking an upgrade. 

This might be decisive in dropping that commit from this PR, let me know.

### CODE REVIEW

**Comment by kailuowang** [comment]
As of now on travis we only run coverage on Scala 2.12 using this [job](https://github.com/typelevel/cats/blob/master/.travis.yml#L32-L35)

With this change we shall be able to delete this coverage dedicated job and run coverage for both 2.12 and 2.13 by changing the jvm tests job to do coverage as well.  I.e. change the following 
https://github.com/typelevel/cats/blob/master/.travis.yml#L47 
to 
```
      install: pip install --user codecov
      script: sbt ++$TRAVIS_SCALA_VERSION!  coverage buildJVM bench/test coverageReport && codecov
```

**Comment by DieBauer** [comment]
@kailuowang thanks for the remark, I forgot about travis. 

I'm reading the PR (https://github.com/typelevel/cats/pull/2737) that added the 2.12 coverage job, and there it states that running coverage reports for multiple scala versions is 'undefined'. 

So I'm not entirely sure if this is what we want to enable here. What are your thoughts?

Also, regarding publishing API docs for 2.12 and 2.13 does anything need to change? I'm not familiar with microsite. In my projects I'm doing `+unidoc` and move the /target/docs/scala-x to a server.

**Comment by kailuowang** [comment]
My thought was enabling coverage on both Scala versions makes some sense. True we will only get one codecov report, but I believe we have coverage threshold check, which fails the build if the delta coverage falls below (on the other hand I haven't seen that in action for a while, so maybe we need to check the settings, not necessarily in this PR though). For those coverage checks it kind of makes sense to have them on both Scala versions. 

Unfortunately our microsite is tied to only one Scala version. I propose we change it to Scala 2.13, although I am not 100% sure if sbt-microsite supports it yet.

**Comment by DieBauer** [comment]
Ok that makes sense. Not sure what the correct place would be in the travis job though

It seems that `bench/test` works on 2.13.0 as well. So the comment (https://github.com/typelevel/cats/blob/master/.travis.yml#L51) is not valid anymore and we can merge the jobs
with something like this?
```yml
    - &jvm_tests
      stage: test
      env: TEST="JVM tests"
      script: sbt ++$TRAVIS_SCALA_VERSION! buildJVM bench/test
      scala:
      - *scala_version_212
      - *scala_version_213
```

Where would coverage reporting then fit in `&& codecov`? in this job, or still in the separate job that you mentioned?

**Comment by kailuowang** [comment]
that's good news. 
yeah, let's put the coverage in the `jvm_tests` jobs

File: build.sbt [hunk 1]
```diff
@@ -6,12 +6,7 @@ import scala.xml.transform.{RewriteRule, RuleTransformer}
 import sbtcrossproject.CrossPlugin.autoImport.{crossProject, CrossType}
 
 lazy val scoverageSettings = Seq(
-  coverageEnabled := {
-    if (priorTo2_13(scalaVersion.value))
-      coverageEnabled.value
-    else
-      false
-  },
+  coverageEnabled := coverageEnabled.value,
```
Comment (kailuowang) [inline build.sbt]: Nit, we don't need this line right?

File: build.sbt [hunk 2]
```diff
@@ -77,11 +72,7 @@ lazy val commonSettings = commonScalaVersionSettings ++ Seq(
     val unchanged = doctestGenTests.value
     if (priorTo2_13(scalaVersion.value)) unchanged else Nil
   },
-  // TODO: re-enable disable scaladoc on 2.13 due to https://github.com/scala/bug/issues/11045
-  sources in (Compile, doc) := {
-    val docSource = (sources in (Compile, doc)).value
-    if (priorTo2_13(scalaVersion.value)) docSource else Nil
-  }
+  sources in (Compile, doc) := (sources in (Compile, doc)).value
```
Comment (kailuowang) [inline build.sbt]: This line is not doing anything.

File: build.sbt (hunk 1 above)
Comment (DieBauer) [reply build.sbt]: Yes correct :)

File: build.sbt (hunk 2 above)
Comment (DieBauer) [reply build.sbt]: Oops, fixed

**Comment by DieBauer** [comment]
Fixed merge conflict with master.

**Pull request merged**